blk/components/virt.c

```diff
@@ -123,12 +123,12 @@ static void partitions_init()
     ((blk_storage_info_t *)blk_config)->sector_size = ((blk_storage_info_t *)blk_config_driver)->sector_size;
```

Comment:
(not for this patch) TODO: refactor to get rid of all the casts.

Reply:
I don't think it's too much of a hassle to do this in this patch, at least for the blk virt.